pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -175,26 +255,98 @@ func (r *ReconcileVirtualMachineImport) Reconcile(request reconcile.Request) (re
-	// Id of the VM specified:
-	if vmSpec.ID != nil {
-		response, err := con.SystemService().VmsService().VmService(*vmSpec.ID).Get().Send()
-		if err != nil {
-			return nil, err
+func ensureDVSecretExists(client client.Client, instance *v2vv1alpha1.VirtualMachineImport, namespace string, ovirtSecretDataMap map[string]string) error {
+	secretObj := &corev1.Secret{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: ovirtSecret, Namespace: namespace}, secretObj)
+	if err != nil && errors.IsNotFound(err) {
+		dvSecret := createDVSecret(ovirtSecretDataMap, instance)
+		return client.Create(context.TODO(), dvSecret)
+	}
+	return err
+}
+
+func ensureDVConfigmapExists(client client.Client, instance *v2vv1alpha1.VirtualMachineImport, namespace string, ovirtSecretDataMap map[string]string) error {
+	cm := &corev1.ConfigMap{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: ovirtConfigmap, Namespace: namespace}, cm)
+	if err != nil && errors.IsNotFound(err) {
+		dvCm := createDVConfigmap(ovirtSecretDataMap, instance)
+		return client.Create(context.TODO(), dvCm)
+	}
+	return err
+}
+
+func createDVConfigmap(creds map[string]string, vmImport *v2vv1alpha1.VirtualMachineImport) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ovirtConfigmap,
+			Namespace: vmImport.Namespace,
+		},
+		Data: map[string]string{
+			"ca.pem": creds["caCert"],
+		},
+	}
+}
+
+func createDVSecret(creds map[string]string, vmImport *v2vv1alpha1.VirtualMachineImport) *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ovirtSecret,
+			Namespace: vmImport.Namespace,
+		},
+		Data: map[string][]byte{
+			"accessKeyId": []byte(creds["username"]),
+			"secretKey":   []byte(creds["password"]),
+		},
+	}
+}
+
+func createVMSpec(vm *ovirtsdk.Vm, vmImport *v2vv1alpha1.VirtualMachineImport) *kubevirtv1.VirtualMachine {
+	cpu := &kubevirtv1.CPU{}
+	if cpuDef, available := vm.Cpu(); available {
+		if topology, available := cpuDef.Topology(); available {
+			if cores, available := topology.Cores(); available {
+				cpu.Cores = uint32(cores)
+			}
+			if sockets, available := topology.Sockets(); available {
+				cpu.Sockets = uint32(sockets)
+			}
+			if threads, available := topology.Threads(); available {
+				cpu.Threads = uint32(threads)
+			}
 		}
-		vm, _ := response.Vm()
-		return vm, nil
 	}
-	// Cluster/name of the VM specified:
-	response, err := con.SystemService().VmsService().List().Search(fmt.Sprintf("name=%v and cluster=%v", *vmSpec.Name, *vmSpec.Cluster.Name)).Send()
-	if err != nil {
-		return nil, err
+	running := false
+	name, _ := vm.Name()
+	if vmImport.Spec.TargetVMName != nil {
+		name = *vmImport.Spec.TargetVMName
 	}
-	vms, _ := response.Vms()
-	if len(vms.Slice()) > 0 {
-		return vms.Slice()[0], nil
+	return &kubevirtv1.VirtualMachine{
```

Comment:
We need to use common template here matched by source operating system or guest operating system. When using OS.Type we need to use pattern matching to understand which template should be used as base for target vm.

Reply:
I think this can be done in separate PR.